components/dashboard/src/components/ItemsList.tsx

```diff
@@ -38,15 +38,17 @@ export function ItemFieldContextMenu(props: {
     menuEntries: ContextMenuEntry[];
     className?: string;
     position?: "start" | "center" | "end";
+    changeMenuState?: (state: boolean) => void;
 }) {
     const cls = "self-" + (props.position ?? "center");
+
     return (
         <div
             className={`flex hover:bg-gray-200 dark:hover:bg-gray-700 rounded-md cursor-pointer w-8 ${cls} ${
                 props.className || ""
             }`}
         >
-            <ContextMenu menuEntries={props.menuEntries} />
+            <ContextMenu changeMenuState={props.changeMenuState} menuEntries={props.menuEntries} />
```

Comment:
**question**: Not sure if this the best line to add this comments but the environment variables list behaves a bit differently without keeping the hover state of the entry when moving the pointer away. Is this something we could fix here or in a separate PR? What do you think?

https://user-images.githubusercontent.com/120486/216107633-71f21f1e-8b11-455c-91de-d26cab5dd0d5.mov

Reply:
Actually, I am waiting for this PR to get reviewed once, so I will get a bit clear with should I setup a separate context for dropdown menu or the functions approach would be good, 
If I need to go with the Separate context approach then all the components would be fixed at once and no extra PR would be needed, but setting up a separate context needs more code changes through the whole tree, 
If I go by the functions approach then I have to make separate PRs for environment variables, integrations and Access Tokens.